activerecord/lib/active_record/errors.rb

```diff
@@ -257,6 +257,19 @@ class ValueTooLong < StatementInvalid
   class RangeError < StatementInvalid
   end
 
+  # Raised when a statement produces an SQL warning.
+  class SQLWarning < StatementInvalid
```

Comment:
This inheritance feels slightly off to me. I'd interpret `StatementInvalid` to mean "the DB rejected this query", which is not the same as an SQLWarning exception, which says that the statement was "bad" but not "invalid". Another difference is that it's probably assumed that `StatementInvalid` implies that the query was not run and if we were in a transaction that it is rolled back, which is not the case with a warning.

I guess the question is whether we want the inheritance or just the behaviour of `StatementInvalid` (having `sql` and `binds`, though it looks like we don't pass those up). If the latter maybe we could introduce a base `StatementError` for both to inherit from, or introduce a module, or have this inherit just ActiveRecordError.

Reply:
Yeah, good points John! 👍 I don't have strong opinions either way -- I don't think we need the inheritance of `StatementInvalid` (as you mentioned, we aren't even passing `sql` up at this point, which I guess we should have been 😆 ). I'm happy to change this to either inherit from a base `StatementError`, or from `ActiveRecordError` itself. If we were to introduce a base `StatementError`, there are maybe other candidates there (e.g. `PreparedStatementInvalid`, which currently inherits from `ActiveRecordError`), but perhaps it's simplest to inherit from the top-level error. I'm open to either!

Reply:
👍 inheriting ActiveRecordError looks great